bundles/org.openhab.binding.xmltv/src/main/java/org/openhab/binding/xmltv/internal/handler/XmlTVHandler.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.xmltv.internal.handler;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.transform.stream.StreamSource;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.xmltv.internal.configuration.XmlTVConfiguration;
+import org.openhab.binding.xmltv.internal.jaxb.Programme;
+import org.openhab.binding.xmltv.internal.jaxb.Tv;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link XmlTVHandler} is responsible for handling XMLTV file and dispatch
+ * information made available to according Media Channels
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class XmlTVHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(XmlTVHandler.class);
+    private final XMLInputFactory xif = XMLInputFactory.newFactory();
+    private final JAXBContext jc;
+
+    private @Nullable Tv currentXmlFile;
+    private @NonNullByDefault({}) ScheduledFuture<?> reloadJob;
+
+    public XmlTVHandler(Bridge thing) throws JAXBException {
+        super(thing);
+        xif.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+        jc = JAXBContext.newInstance(Tv.class);
+    }
+
+    @Override
+    public void initialize() {
+        XmlTVConfiguration config = getConfigAs(XmlTVConfiguration.class);
+        logger.debug("Initializing {} for input file '{}'", getClass(), config.filePath);
+
+        reloadJob = scheduler.scheduleWithFixedDelay(() -> {
+            final StreamSource source = new StreamSource(config.filePath);
+            currentXmlFile = null;
+            XMLStreamReader xsr = null;
+            try {
```

Comment:
You can probably simplify this job by using the `XMLStreamReader` with a [try-with-resources statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html).

Reply:
Did not know it. Thanks but apparently not, XMLStreamReader does not implement AutoCloseable

Reply:
Thanks for looking into it! Too bad it's not implemented. :disappointed: